.github/scripts/collect_github_metrics.py

```diff
@@ -0,0 +1,175 @@
+
+    # Create tables
+    cur = conn.cursor()
+    create_db_tables(conn, cur)
+
+    # Get the data
+    g = Github(github_token)
+    repo = g.get_repo(repo_name)
+
+    run = repo.get_workflow_run(int(run_id))
+
+    workflow_data_query = f'''INSERT INTO github_workflow_runs_test(
+    run_id, html_url, name,
+    run_started_at, triggering_actor_login, conclusion,
+    run_number, event, run_attempt, repository_full_name,
+    head_branch, display_title, path)
+    VALUES(
+    '{run_id}', '{run.html_url}', '{run.name}', '{run.run_started_at}',
+    '{run.raw_data['triggering_actor']['login']}',
+    '{run.conclusion}', '{run.run_number}', '{run.event}',
+    '{run.run_attempt}', '{run.raw_data['repository']['full_name']}',
+    '{run.head_branch}', '{run.display_title}', '{run.path}'
+    );
+    '''
+
+    logger.debug('Workflow run query: %s', workflow_data_query)
+    cur.execute(workflow_data_query)
+
+    for job in run.jobs():
+        job_id = job.id
+        queued_duration_seconds = 0
+        duration_seconds = 0
+
+        job_created_at_date = dateutil.parser.parse(job.raw_data['created_at'])
+
+        queued_duration_timedelta = job.started_at - job_created_at_date
+        queued_duration_seconds = round(queued_duration_timedelta.total_seconds())
+
+        duration_timedelta = job.completed_at - job.started_at
+        duration_seconds = round(duration_timedelta.total_seconds())
+
+        job_data_query = f'''
+        INSERT INTO github_workflow_jobs_test(
+        job_id, parent_run_id, html_url, name,
+        created_at, started_at, completed_at,
+        queued_duration_seconds, duration_seconds,
+        runner_name, status, conclusion, head_branch)
+        VALUES(
+        '{job_id}', '{run_id}', '{job.html_url}', '{job.name}',
+        '{job.raw_data['created_at']}', '{job.started_at}', '{job.completed_at}',
+        '{queued_duration_seconds}', '{duration_seconds}',
+        '{job.raw_data['runner_name']}', '{job.status}', '{job.conclusion}',
+        '{job.raw_data['head_branch']}'
+        );
+        '''
+        logger.debug('Job query: %s', job_data_query)
+        cur.execute(job_data_query)
+        for step in job.steps:
+            duration_seconds_timedelta = step.completed_at - step.started_at
+            duration_seconds = round(duration_seconds_timedelta.total_seconds())
+
+            step_data_query = f'''
+            INSERT INTO github_workflow_steps_test(
+            parent_job_id, name, conclusion,
+            number, started_at, completed_at,
+            duration_seconds)
+            VALUES(
+            '{job_id}', '{step.name}','{step.conclusion}',
+            '{step.number}', '{step.started_at}', '{step.completed_at}',
+            '{duration_seconds}'
+            );
+            '''
+            logger.debug('Step query: %s', step_data_query)
+            cur.execute(step_data_query)
+
+    conn.commit()
+    cur.close()
+    conn.close()
+    g.close()
+if __name__ == "__main__":
```

Comment:
maybe use `ArgumentParser` from `argparse` so that the script could be run like `python3 collect_github_metrics.py --run_id <> ...`? 
It could ease the debugging/running and provide automatic help creation for the script. The env var approach seems a bit odd from an outside perspective.

It could be done later though.

Reply:
The first plan was to turn this script to a GitHub Action, that's why environment variables have been chosen. Right now it seems like we should do both, but on the next stage indeed.